### DISCUSSION TITLE
When creating new notes with denote, the echo area shows a prompt "Non-existent agenda file _org-file-name_.  [R]emove from list or [A]bort?"

### DISCUSSION THREAD

**Discussion by IceAsteroid** [protesilaos/denote#433] [Q&A] [answered]
It only happens when creating org files when `denote`, the echo area prompts the message `Non-existent agenda file ~/OrgWiki_Test/20240913T111111--TestOrgFile__emacs.org.  [R]emove from list or [A]bort?`.

It's waiting the user to response to it, before the created org file is switched to.

`When` I use `find-file` to create or open an org file, this issue doesn't occor.

And the message buffer outputs:

```
Non-existent agenda file ~/OrgWiki_Test/20240913T111111--TestOrgFile__emacs.org.  [R]emove from list or [A]bort?
File mode specification error: (user-error Abort)
```

This happens when I have a function that invokes `org-map-entries` and have it added to the `org-mode-hook`.

When I remove it from the hook, the issue disappears.

Relevant issue post on this: https://www.reddit.com/r/emacs/comments/1bnhz24/bug_nonexistent_agenda_file_s/

I don't know this is either the issue of `org-mode`, `denote`, or just the function I've defined, the issue post said that the issue is fixed, but it still occurs in this case

Here's the code that's related to the function:

```
;; Implement subentry count in title for each heading
;;-This implementation was created as barely a scratch by me
;;-, but was improved to perfection by @dalanicolai on StackExchange.
;;- https://emacs.stackexchange.com/questions/79785/how-to-display-sub-heading-count-somewhere-in-a-heading-in-org-mode/79829?noredirect=1#comment133905_79829
(defun my/org-subentry-count ()
  (defcustom org-subentry-count-at-start nil
    "Show the subentry count before the title, not after."
    :group 'org-subenty-count
    :type 'boolean)
  (defvar org-subentry-count-in-headings-p nil
    "t to enable org-subentry-count-in-headings
       nil to disable org-subentry-count-in-headings")
  (defun org-subentry-count-in-headings ()
    (org-map-entries
     (lambda ()
       (when (looking-at org-complex-heading-regexp)
	 (let* ((group (if org-subentry-count-at-start 1 0))
		(beg (match-beginning group))
		(end (match-end group))
		(level (length (match-string 1))) ; stars in headline
		(subheading-count (format "%s" (1- (length (org-map-entries nil nil 'tree)))))
		(ov (car (seq-filter
			  ;; search for subentry-count overlay
			  (lambda (o)
			    (eq (overlay-get o 'type) 'subentry-count)) 
			  (overlays-in (line-beginning-position)
				       (line-end-position))))))
	   (if ov ; only create overlay if not exists yet
	       (move-overlay ov beg end)
	     (setq ov (make-overlay beg end))
	     (overlay-put ov 'type 'subentry-count) ; add a type, for easy filtering
	     (overlay-put ov 'evaporate t))
	   ;; only add/remove 'after-string' prop in toggle
	   (if org-subentry-count-in-headings-p
	       (overlay-put ov 'after-string (propertize (format " [%s]" subheading-count)
							 'face (intern-soft (format "org-level-%d" level))))
	     (overlay-put ov 'after-string nil)))))))
  (defun my/toggle-org-subentry-count-in-headings ()
    (interactive)
    (if org-subentry-count-in-headings-p
	(setq org-subentry-count-in-headings-p nil)
      (setq org-subentry-count-in-headings-p t))
    (org-subentry-count-in-headings))
  (defun my/toggle-org-subentry-position ()
    (interactive)
    (setq org-subentry-count-at-start (not org-subentry-count-at-start))
    (setq org-subentry-count-in-headings-p t)
    (org-subentry-count-in-headings))
  ;; Trying to fix the org-agenda-files error, but failed.
  (defun my/enable-org-subentry-count-in-headings ()
    (when (buffer-match-p ".*\.org" (current-buffer))
      (org-subentry-count-in-headings)))
  (add-hook 'org-mode-hook (lambda ()
			     (setq org-subentry-count-in-headings-p t)
			     (setq org-subentry-count-at-start t)
			     ;; (my/enable-org-subentry-count-in-headings)
			     (org-subentry-count-in-headings)
			     (add-hook 'after-save-hook 'org-subentry-count-in-headings)
			     ))
  )
(my/org-subentry-count)
```

If this is not caused by `denote`, please tell me that, and I'll post it on org-mode repo

**Comment by IceAsteroid** [comment]
I didn't configure any agenda related settings in my configuration file. I don't what caused this..

Here's the demo:

https://github.com/user-attachments/assets/2e225330-5573-43b6-9e43-fa4d4697deb9

**Comment by protesilaos** [comment]
> From: IceAsteroid ***@***.***>
> Date: Fri, 13 Sep 2024 05:43:07 -0700
>
> It only happens when creating org files when `denote`, when I use
> `find-file` to create or open an org file, this issue doesn't occor.
>
> And the message buffer outputs:
>
> ```
> Non-existent agenda file ~/OrgWiki_Test/20240913T111111--TestOrgFile__emacs.org.  [R]emove from list or [A]bort?
> File mode specification error: (user-error Abort)
> ```

This error means that at some point you had that TestOrgFile in your
'org-agenda-files', but it is not there anymore.

Try to explicitly set the value of 'org-agenda-files' and see what
happens. I do not think this is related to Denote.

Examples:

    ;; Include only those files.
    (setq org-agenda-files '("/path/to/file/one.org" "/path/to/file/two.org"))

    ;; Include all files inside of the `org-directory'.
    (setq org-agenda-files (list org-directory))

-- 
Protesilaos Stavrou
https://protesilaos.com

**Reply by IceAsteroid** [reply]
The issue still occurs, I'll try something else. But anyway, thank you for the help :)

**Comment by IceAsteroid** [comment, marked as answer]
Good news, the following code as I linked in the description works

The issue was triggered by `org-map-entries`.

Here's the code by a deleted user in reddit:

----

> I don't know this is either the issue of org-mode, denote, or just the function I've defined, the issue post said that the issue is fixed, but it still occurs in this case

**Note**: If someone has encountered this issue: It's because the latest version of org.el still doesn't contain the fix as mentioned in the link on reddit, you need to use the following code as a workaround.

```
(defun my-org-agenda-prepare-buffers-advice (orig-func &rest args)
  "Advice function to modify `org-agenda-prepare-buffers'.
It filters the FILES argument to ensure only agenda files are processed."
  (let ((files (if (listp (car args)) (car args) nil)))
    (when files
      (setq files (seq-filter #'(lambda (file) (member file org-agenda-files)) files)))
    (apply orig-func (list files))))

(advice-add 'org-agenda-prepare-buffers :around #'my-org-agenda-prepare-buffers-advice)

```
----

Sorry for the noise.
